src/main/java/org/springframework/retry/annotation/AnnotationAwareRetryOperationsInterceptor.java

```diff
@@ -451,7 +452,8 @@ private BackOffPolicy getBackoffPolicy(Backoff backoff, boolean stateless) {
 		boolean isRandom = false;
 		String randomExpression = (String) attrs.get("randomExpression");
 		Expression parsedRandomExp = null;
-		if (multiplier > 0) {
+
+		if (getMultiplier(multiplier, parsedMultExp, stateless) > 0) {
```

Comment:
This is still not correct: the `multiplier` might be calculated at runtime against each call.
That's exactly the point of those SpEL expressions.
I suggested before that this `random` is calculated unconditionally.

See the logic in the `BackOffPolicyBuilder`:
```
if (this.multiplier != null && this.multiplier > 0 || this.multiplierSupplier != null) {
```
So, if runtime expression is provided (read `multiplierSupplier`), then we don't care to what value it is going to be resolved.
We just cannot always resolve `parsedMultiplierExpression` at configuration phase since it might not be end-user intention.
Therefore that new `getMultiplier()` must be removed.

Reply:
Ok. Sorry for the confusion. Got your point now

Reply:
Please check now